### PR TITLE
Pseduo-remove MapToCurve::new, renamed to check_parameters

### PR DESCRIPTION
**Pull request by burdges**
## Description

MapToCurve::new seemingly originates from a more runtime oriented elliptic curve crate: https://github.com/armfazh/redox-ecc/blob/master/src/ellipticcurve.rs#L36
    
Arguably test_parameters should be inherent methods, invoked by whoever defined the parameters, but maybe not?  I left the trait method for now.  I suspect the hash-to-curve code would benefit form a wider more idiomatic rewrite btw.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

File: ec/src/hashing/curve_maps/swu.rs [hunk 1]
```diff
@@ -35,8 +35,8 @@ pub fn parity<F: Field>(element: &F) -> bool {
 }
 
 impl<P: SWUConfig> MapToCurve<Projective<P>> for SWUMap<P> {
-    /// Constructs a new map if `P` represents a valid map.
-    fn new() -> Result<Self, HashToCurveError> {
+    /// Checks if `P` represents a valid map.
+    fn test_parameters() -> Result<(), HashToCurveError> {
```
Comment (Pratyush) [inline ec/src/hashing/curve_maps/swu.rs:39]:
```suggestion
    fn check_parameters() -> Result<(), HashToCurveError> {
```

File: ec/src/hashing/curve_maps/wb.rs [hunk 2]
```diff
@@ -79,13 +79,13 @@ pub trait WBConfig: SWCurveConfig + Sized {
...
-    swu_field_curve_hasher: SWUMap<P::IsogenousCurve>,
+    swu_field_curve_hasher: PhantomData<SWUMap<P::IsogenousCurve>>,
     curve_params: PhantomData<fn() -> P>,
 }
 
 impl<P: WBConfig> MapToCurve<Projective<P>> for WBMap<P> {
-    /// Constructs a new map if `P` represents a valid map.
-    fn new() -> Result<Self, HashToCurveError> {
+    /// Checks if `P` represents a valid map.
+    fn test_parameters() -> Result<(), HashToCurveError> {
```
Comment (Pratyush) [inline ec/src/hashing/curve_maps/wb.rs:88]:
```suggestion
    fn check_parameters() -> Result<(), HashToCurveError> {
```

File: ec/src/hashing/curve_maps/swu.rs [hunk 3]
```diff
@@ -256,14 +256,12 @@ mod test {
     /// elements should be mapped to curve successfully. everything can be mapped
     #[test]
     fn map_field_to_curve_swu() {
-        let test_map_to_curve = SWUMap::<TestSWUMapToCurveConfig>::new().unwrap();
+        SWUMap::<TestSWUMapToCurveConfig>::test_parameters().unwrap();
```
Comment (Pratyush) [inline ec/src/hashing/curve_maps/swu.rs:259]:
```suggestion
        SWUMap::<TestSWUMapToCurveConfig>::check_parameters().unwrap();
```

File: ec/src/hashing/map_to_curve_hasher.rs [hunk 4]
```diff
@@ -4,11 +4,11 @@ use ark_std::marker::PhantomData;
 
 /// Trait for mapping a random field element to a random curve point.
 pub trait MapToCurve<T: CurveGroup>: Sized {
-    /// Constructs a new mapping.
-    fn new() -> Result<Self, HashToCurveError>;
+    /// Checks whether supplied parameters represent a valid map.
+    fn test_parameters() -> Result<(), HashToCurveError>;
```
Comment (Pratyush) [inline ec/src/hashing/map_to_curve_hasher.rs:8]:
```suggestion
    fn check_parameters() -> Result<(), HashToCurveError>;
```

**Review by Pratyush — APPROVED** [review]

**Review by mmagician — APPROVED** [review]
Looks good. @burdges Can you merge latest arkworks master into your branch pls? I can't do it since the PR came from an org

**Comment by burdges** [comment]
Appears @weikengchen did so, no idea how these rules all work.

Just fyi, I've a larger successor https://github.com/arkworks-rs/algebra/compare/master...w3f:arkworks-algebra:xof_h2c approaching readiness, which answers #629 and #630.

**Comment by mmagician** [comment]
Seems that the previous merge introduced some conflicts. Maybe that's a good chance for @burdges to add a line to the CHANGELOG :)

**Comment by Pratyush** [comment]
I have one question @burdges: the current approach ensures that one can't construct an invalid map and use it, while the new code allow using an invalid mapping. Do you think there's an easy way to preserve the old behaviour?

**Comment by burdges** [comment]
> Do you think there's an easy way to preserve the old behaviour?

It could be invoked behind `#[cfg(debug_assertions)]` further up the stack: https://github.com/arkworks-rs/algebra/compare/master...w3f:arkworks-algebra:xof_h2c?expand=1#diff-d4ed998b41839f70c99ab6471a8c55b448ae708642d0a8e87f826e4b071e79f8R54

**Comment by burdges** [comment]
I'll close this in favor of #679 which is basically identical.

**Pull request closed without merging**